src/reader/mod.rs

```diff
@@ -143,7 +191,19 @@ impl InnerIndexReader {
         })
         .take(self.num_searchers)
         .collect::<io::Result<_>>()?;
+        let warmers = self.list_warmers();
+        self.warming_executor
+            .map(|warmer| warmer.warm(&searchers[0]), warmers.iter().cloned())?;
         self.searcher_pool.publish_new_generation(searchers);
```

Comment:
I would ideally like to ensure that old `Searcher`s have been dropped, so that we are not calling `retain()` early while there are still active searchable segments with previously leased searchers. Not sure how to architect that.

this test tweak fails: https://gist.github.com/314e5029010c42c4e39f9d91da4c7a1f

Reply:
Addressed in https://github.com/quickwit-inc/tantivy/pull/1246/commits/c9346cbf8aaf4160c1276bc81b5f5353bbf55ee0